### DISCUSSION TITLE
Get-PnPProperty : Operation is not valid due to the current state of the object

### DISCUSSION THREAD

**Discussion by sgray563** [pnp/powershell#3079] [General]
Hi all, I've been working on a PS script to find files in a SharePoint site and library larger than a given size and check each for versions and total version size. When trying to retrieve the versions of some files I receive an error: 

Get-PnPProperty : Operation is not valid due to the current state of the object.

When I look at these files in SharePoint I can see there are versions. Problem files seem to be greater than 2GB but not all over 2GB give the error (presumably they don't have versions). Can someone please advise where I might make a change to retrieve the versions and version size on these problem files?

```
$Items = Get-PnPListItem -List $List -PageSize 500 | Where {($_.FieldValues.FileLeafRef -like "*.*") -and ($_.FieldValues.Created -gt $FilterDate) -and ($_.FieldValues.SMTotalFileStreamSize/1MB -gt 1000)}

ForEach ($Item in $Items)
{
    $FileSize = ([math]::Round(($Item.FieldValues.SMTotalFileStreamSize/1MB),2))

    $File = Get-PnPProperty -ClientObject $Item -Property File
    $Versions = Get-PnPProperty -ClientObject $File -Property Versions 
    
    #If the file has other versions sum their size
    If ($Versions.Count -gt 1) {
        $VersionSize = $Versions | Measure-Object -Property Size -Sum | Select-Object -Expand Sum
        $VersionSize = [Math]::Round(($VersionSize/1MB),2)
        $TotalFileSize = [Math]::Round(($FileSize + $VersionSize),2)
    } Else {
        $VersionSize = 0
        $TotalFileSize = [Math]::Round(($FileSize + $VersionSize),2)
    }
    
    $Results += [PSCustomObject][ordered]@{
        Library         = $List.Title
        URL             = $Item.FieldValues.FileRef
        Versions        = $Versions.Count 
        VersionSize     = $VersionSize
        TotalFileSize   = $TotalFileSize
    }
}

#Export the results to CSV
$Results | Export-Csv -Path $FileName -NoTypeInformation
```

**Comment by sgray563** [comment]
Further to this, my colleagues and I have been testing with Get-PnPProperty and Get-PnPFileVersion and both throw the same error when attempting to retrieve versions greater than 2GB. Similarly trying to delete versions using Remove-PnPFileVersion fails with the same error. 

There is a similar discussion here (https://github.com/pnp/powershell/discussions/2696) but it's now closed. Can anyone advise if this is a bug that's being looked at or if it's a user error? If there is some other way to get and delete versions greater than 2GB please let me know.

**Comment by alekpavlov** [comment]
I'm also facing the same issue: unable to delete with **Remove-PnPFileVersion** version of a file which is 9 GB. Do any of you found a solution or workaround to this problem?

**Comment by YuriySamorodov** [comment]
Same issue here as well

**Comment by waaromikniet** [comment]
Maybe it has something to do with the HTTP Timeout setting in the PnP.Core library?
https://pnp.github.io/pnpcore/using-the-sdk/files-large.html

**Comment by mark74univpm** [comment]
Is there any solution? I'm facing the very same problem
